src/promise.ts

```diff
@@ -1,228 +1,349 @@
-import { Bytes } from "./utils";
+import { Bytes, PromiseIndex } from "./utils";
 import * as near from "./api";
 import { Balance, PublicKey, AccountId, Gas, GasWeight } from "./types";
 import { Nonce } from "./types/primitives";
 
 export abstract class PromiseAction {
-    abstract add(promise_index: number | bigint): void;
+  abstract add(promiseIndex: PromiseIndex): void;
 }
 
 export class CreateAccount extends PromiseAction {
-    add(promise_index: number | bigint) {
-        near.promiseBatchActionCreateAccount(promise_index);
-    }
+  add(promiseIndex: PromiseIndex) {
+    near.promiseBatchActionCreateAccount(promiseIndex);
+  }
 }
 
 export class DeployContract extends PromiseAction {
-    constructor(public code: Bytes) { super() }
+  constructor(public code: Bytes) {
+    super();
+  }
 
-    add(promise_index: number | bigint) {
-        near.promiseBatchActionDeployContract(promise_index, this.code);
-    }
+  add(promiseIndex: PromiseIndex) {
+    near.promiseBatchActionDeployContract(promiseIndex, this.code);
+  }
 }
 
-export class FunctionCall extends PromiseAction{
-    constructor(public function_name: string, public args: Bytes, public amount: Balance, public gas: Gas) { super() }
-
-    add(promise_index: number | bigint) {
-        near.promiseBatchActionFunctionCall(promise_index, this.function_name, this.args, this.amount, this.gas);
-    }
+export class FunctionCall extends PromiseAction {
+  constructor(
+    public functionName: string,
+    public args: Bytes,
+    public amount: Balance,
+    public gas: Gas
+  ) {
+    super();
+  }
+
+  add(promiseIndex: PromiseIndex) {
+    near.promiseBatchActionFunctionCall(
+      promiseIndex,
+      this.functionName,
+      this.args,
+      this.amount,
+      this.gas
+    );
```

Comment:
My personal feel is above one line isn't too long to be unreadable, but this become kind of lengthy. What is the print width we had for other NEAR JS projects?

Reply:
@ailisp I haven't set any of those preferences myself, just used `prettier` defaults, so we can change this to anything really.